diff -Naur libvdpau-0.9/src/Makefile.am libvdpau-0.9.tpg/src/Makefile.am
--- libvdpau-0.9/src/Makefile.am	2014-11-05 00:17:39.000000000 +0000
+++ libvdpau-0.9.tpg/src/Makefile.am	2014-12-21 00:57:59.414244701 +0000
@@ -22,7 +22,7 @@
 libvdpau_la_LIBADD = \
     $(DLOPEN_LIBS) \
     $(PTHREAD_LIBS) \
-    $(XEXT_LIBS)
+    $(XEXT_LIBS) -lX11
 
 libvdpau_la_LDFLAGS = -version-info 1:0:0 -no-undefined
 
diff -Naur libvdpau-0.9/src/Makefile.in libvdpau-0.9.tpg/src/Makefile.in
--- libvdpau-0.9/src/Makefile.in	2014-12-19 17:33:35.000000000 +0000
+++ libvdpau-0.9.tpg/src/Makefile.in	2014-12-21 00:58:08.960500858 +0000
@@ -357,7 +357,7 @@
 libvdpau_la_LIBADD = \
     $(DLOPEN_LIBS) \
     $(PTHREAD_LIBS) \
-    $(XEXT_LIBS)
+    $(XEXT_LIBS) -lX11
 
 libvdpau_la_LDFLAGS = -version-info 1:0:0 -no-undefined
 libvdpauincludedir = $(includedir)/vdpau
