diff -Naur libvdpau-0.4.1//src/Makefile.am libvdpau-0.4.1.tpg//src/Makefile.am
--- libvdpau-0.4.1//src/Makefile.am	2010-09-07 17:21:32.000000000 +0000
+++ libvdpau-0.4.1.tpg//src/Makefile.am	2010-09-12 16:01:01.000000000 +0000
@@ -18,7 +18,7 @@
 
 libvdpau_la_LIBADD = \
     $(DLOPEN_LIBS) \
-    $(XEXT_LIBS)
+    $(XEXT_LIBS) -lX11
 
 libvdpau_la_LDFLAGS = -version-info 1:0:0 -no-undefined
 
diff -Naur libvdpau-0.4.1//src/Makefile.in libvdpau-0.4.1.tpg//src/Makefile.in
--- libvdpau-0.4.1//src/Makefile.in	2010-09-08 17:07:03.000000000 +0000
+++ libvdpau-0.4.1.tpg//src/Makefile.in	2010-09-12 16:01:28.000000000 +0000
@@ -234,7 +234,7 @@
     -I$(top_srcdir)/include \
     -DVDPAU_MODULEDIR="\"$(moduledir)\"" \
     $(X11_CFLAGS) \
-    $(XEXT_CFLAGS)
+    $(XEXT_CFLAGS) -lX11
 
 lib_LTLIBRARIES = libvdpau.la
 libvdpau_la_SOURCES = \
